bundles/org.smarthomej.binding.notificationsforfiretv/src/main/java/org/smarthomej/binding/notificationsforfiretv/internal/NotificationsForFireTVThingActions.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ * Copyright (c) 2021 Contributors to the SmartHome/J project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.smarthomej.binding.notificationsforfiretv.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.automation.annotation.ActionInput;
+import org.openhab.core.automation.annotation.ActionOutput;
+import org.openhab.core.automation.annotation.RuleAction;
+import org.openhab.core.thing.binding.ThingActions;
+import org.openhab.core.thing.binding.ThingActionsScope;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Some automation actions to be used with a {@link NotificationsForFireTVThingActions}
+ *
+ * @author Tom Blum - Initial contribution
+ */
+@ThingActionsScope(name = "notificationsforfiretv")
+@NonNullByDefault
+public class NotificationsForFireTVThingActions implements ThingActions {
+
+    private final Logger logger = LoggerFactory.getLogger(NotificationsForFireTVThingActions.class);
+
+    private @Nullable NotificationsForFireTVHandler handler;
+
+    @RuleAction(label = "Notifications For Fire TV sendNotification", description = "Action that sends notification to Fire TV")
+    public @ActionOutput(name = "success", type = "java.lang.Boolean") Boolean sendNotification(
+            @ActionInput(name = "message") @Nullable String msg,
+            @ActionInput(name = "icon") @Nullable String filename) {
+        final NotificationsForFireTVHandler handler = this.handler;
+        if (handler == null) {
+            logger.warn("Handler is null, cannot send notification.");
+            return false;
+        } else {
+            return handler.sendNotification(msg, filename, null);
+        }
+    }
+
+    @RuleAction(label = "Notifications For Fire TV sendNotification", description = "Action that sends notification to Fire TV")
+    public @ActionOutput(name = "success", type = "java.lang.Boolean") Boolean sendNotification(
```

Comment:
Unfortunately two actions with different paramater set but the same name cannot be used due to limitations in the openHAB framework. Please choose a different name for one of the actions.

Reply:
Sadly i do not understand the problem. I did a test with method 1 (two parameters) and a test with method 2 (three parameters) and both work like they should.

Reply:
I guess it's only a problem in some cases, probably not for Rules DSL. The `AnnotatedThingActionModuleTypeProvider` uses the action/method name and the thing UID to construct the module UID. This is then stored in a `Map` and thus two actions with the same name are not possible. Did you try if you can see both actions when creating rules in the UI?

Reply:
I never created rules in the UI.
Should i change the name sendNotification to sendNotificationWithImage?
If you say there could be a problem, i will change it.

Reply:
That sounds good!

Reply:
done, changed sendNotification to sendNotificationWithImage